LineChart.js

```diff
@@ -0,0 +1,1161 @@
+        plot.select('.x_axis')
+            .transition()
+            .duration(1000)
+            .call(d3.axisBottom(x));
+
+        plot.select('.y_axis')
+            .transition()
+            .duration(1000)
+            .call(d3.axisLeft(y).ticks(null, "s"));
+
+        UTIL.setAxisColor(svg, _yAxisColor, _xAxisColor, _showYaxis, _showXaxis);
+        UTIL.displayThreshold(threshold, data, keys);
+
+    }
+    chart._getName = function () {
+        return _NAME;
+    }
+
+    var _mergeForTransition = function (fData, sData) {
+        var secondSet = d3.set();
+
+        sData.forEach(function (d) {
+            secondSet.add(d[_dimension[0]]);
+        });
+
+        var onlyFirst = fData.filter(function (d) {
+            return !secondSet.has(d[_dimension[0]]);
+        })
+            .map(function (d) {
+                var obj = {};
+
+                obj[_dimension[0]] = d[_dimension[0]];
+                obj[_measure[0]] = 0;
+
+                return obj;
+            });
+
+        return d3.merge([sData, onlyFirst])
+            .sort(function (a, b) {
+                return a[_measure[0]] > b[_measure] ? _sort
+                    : a[_measure[0]] < b[_measure] ? -_sort
+                        : 0;
+            })
+    }
+
+
+    chart.config = function (value) {
+        if (!arguments.length) {
+            return _config;
+        }
+        _config = value;
+        _setConfigParams.call(chart, _config);
+        return chart;
+    }
+
+    chart.dimension = function (value) {
+        if (!arguments.length) {
+            return _dimension;
+        }
+        _dimension = value;
+        return chart;
+    }
+
+    chart.measure = function (value) {
+        if (!arguments.length) {
+            return _measure;
+        }
+        _measure = value;
+        return chart;
+    }
+
+    chart.showLegend = function (value) {
+        if (!arguments.length) {
+            return _showLegend;
+        }
+        _showLegend = value;
+        return chart;
+    }
+
+    chart.legendPosition = function (value) {
+        if (!arguments.length) {
+            return _legendPosition;
+        }
+        _legendPosition = value;
+        return chart;
+    }
+
+    chart.showValueAs = function (value) {
+        if (!arguments.length) {
+            return _showValueAs;
+        }
+        _showValueAs = value;
+        return chart;
+    }
+
+    chart.valueAsArc = function (value) {
+        if (!arguments.length) {
+            return _valueAsArc;
+        }
+        _valueAsArc = value;
+        return chart;
+    }
+
+    chart.valuePosition = function (value) {
+        if (!arguments.length) {
+            return _valuePosition;
+        }
+        _valuePosition = value;
+        return chart;
+    }
+
+    chart.sort = function (value) {
+        if (!arguments.length) {
+            return _sort;
+        }
+        _sort = value;
+        return chart;
+    }
+
+    chart.filterData = function (value) {
```

Comment:
Let's not have this kind of accessor function. Try using a private function instead for these